packages/react-reconciler/src/ReactFiberCompleteWork.new.js

```diff
@@ -223,7 +227,11 @@ if (supportsMutation) {
     let node = workInProgress.child;
     while (node !== null) {
       if (node.tag === HostComponent || node.tag === HostText) {
-        appendInitialChild(parent, node.stateNode);
+        if (supportsHydration && node.flags & Placement) {
```

Comment:
All of these are forked on supportsHydration but these don't have anything to do with hydration do they? E.g. I can render into document.body with createRoot and it's still relevant right?

Reply:
A cheat technique for now to opt out in non dom renderers. I can move to it’s own thing

Reply:
Yeah this does not depend on concurrent roots

Reply:
I think this will all need to go behind a feature flag because it's technically a breaking change. Doesn't this basically implement the whole new `<head>` semantics already except it also unmounts it?